elf/elf.go

```diff
@@ -257,7 +322,21 @@ func elfReadMaps(file *elf.File) (map[string]*Map, error) {
 			mapCount := len(data) / C.sizeof_struct_bpf_map_def
 			for i := 0; i < mapCount; i++ {
 				pos := i * C.sizeof_struct_bpf_map_def
-				cm, err := C.bpf_load_map((*C.bpf_map_def)(unsafe.Pointer(&data[pos])))
+				mapDef := (*C.bpf_map_def)(unsafe.Pointer(&data[pos]))
+
+				var mapPathC *C.char
+				if mapDef.pinning > 0 {
+					mapPath, err := prepareBPFFS(C.GoString(&mapDef.namespace[0]), name)
```

Comment:
You could verify somewhere that `&mapDef.namespace[0]` is really NULL-terminated before the end of the 256-sized buffer, in case someone builds a ELF file with a corrupted `mapDef.namespace`.

Also, you could check that the namespace:
- does not contain any `/ character
- is not empty (equals to "")
- is not equal to "." or ".."